### DISCUSSION TITLE
updated property after js not work

### DISCUSSION THREAD

**Discussion by mckaragulle** [livewire/livewire#8674] [1. Help]
hi there, i have two select2 boxes. first select2 box name is "dealer_id", second select2 box name is "department_id".  When I update the selectbox named dealer_id, the options of the selectbox named department_id are updated. But js is broken. Even though I used the code below to solve this, select2 named department_id does not work.

This code block works on livewire2. But why doesn't it work in the livewire3 version? .select2(); The method behaves as if it is not loaded.

```
window.addEventListener('updatePerson', event => {
      $('.select2').select2();
  });
```

![image](https://github.com/user-attachments/assets/7345d8ab-af3e-4907-b682-8c29ddae93e1)
![image](https://github.com/user-attachments/assets/8892eaa8-d842-4b2b-ab13-2d50c5b0486d)
![image](https://github.com/user-attachments/assets/33d0767b-2d6b-4d3d-94b1-c2d91d70c9ac)
![image](https://github.com/user-attachments/assets/de6d52dc-e7f4-4085-ae21-ee14f63073f1)
![image](https://github.com/user-attachments/assets/947cba2e-a51c-4cfc-91f0-5280ccb83fea)

**Comment by EdgarsJoja** [comment]
Did you try removing document.ready listener?
https://livewire.laravel.com/docs/javascript#executing-scripts

Perhaps in v3, when your script is injected, the document.ready is no longer dispatched (because it might have happened already before the @script part is loaded).

**Reply by mckaragulle** [reply]
Yes I tried. It didn't work again.
It seems very strange to me that this code works in livewire 2 version but does not work in livewire 3 version.

Uploading Ekran Kaydı 2024-07-23 10.45.17.mov…
